### DISCUSSION TITLE
Uncompress Zlib compressed BLOB

### DISCUSSION THREAD

**Discussion by otofoto** [pawelsalawa/sqlitestudio#5063] [General]
How to uncompress Zlib compressed BLOB data? 
I loaded C:/Program Files/SQLiteStudio/ext/compress.dll extension but data still shows as compressed in HEX view!?

**Comment by pawelsalawa** [comment]
From the extension's documentation:
> This SQLite extension implements SQL compression functions compress() and uncompress() using ZLIB.

To use the extension, you need to use these functions to extract data from certain columns:

`SELECT uncompress(compressed_column) FROM the_table;`

**Comment by otofoto** [comment]
Can it be applied to HEX view?
Similar like it is in [SS viewer](https://www.mitec.cz/ssv.html) for structured storage.
![SSView_007 1](https://github.com/user-attachments/assets/ae8e5c13-c97f-48d9-a2d1-8ac38f1ea202)
![SSView_008 1](https://github.com/user-attachments/assets/8ce6fe46-e38f-4332-9feb-73522d1f9542)

**Reply by pawelsalawa** [reply]
If you execute the `SELECT` query with `uncompress()` function applied to a proper column, then you can view results in whatever editor you like (among available editors in SQLiteStudio).

**Reply by otofoto** [reply]
I want to view it here.
![image](https://github.com/user-attachments/assets/7588c127-9ff8-43c9-90fe-57d2a8d42363)

SELECT doesn't seem to work
![image](https://github.com/user-attachments/assets/3b0a16aa-6bb1-46be-a976-ddc6c6d0784b)

**Reply by pawelsalawa** [reply]
Grid view will not show it to you. You need to switch to the "Form view":
![image](https://github.com/user-attachments/assets/7193a260-01ee-429e-9d62-e781ea22e531)

Although the data is NULL, so I don't expect to see anything. Apparently your data in `PointData` is not compatible with `compress` extension.

**Comment by otofoto** [comment]
Seems it is related to this. 

`**   *  This utility uses the "zlib format" like SQLAR, but adds the variable-
**      length integer uncompressed size value at the beginning.`

Normally zlib starts with these numbers not some uncompressed size value.

zlib magic headers
78 01 - No Compression/low
78 5E - Fast Compression
78 9C - Default Compression
78 DA - Best Compression

**Reply by tuffnatty** [reply]
You can try the following:
1. Ensure that ScriptingPython SQLiteStudio plugin is enabled and (if you're using Git HEAD) configured properly.
2. Add a Python function in SQL Functions Manager:

    import zlib
    return zlib.decompress(args[0]) 

4. Use it.

**Reply by otofoto** [reply]
Sounds promising but it is of no use.

`Cannot load plugin ScriptingPython. Error details: Cannot load library C:\Program Files\SQLiteStudio\plugins\ScriptingPython.dll: The specified module could not be found.`

File exists so no idea why it shows error that no file is found.

**Reply by tuffnatty** [reply]
It can't find the Python library (named probably python39.dll on Windows).
So, one option is to install Python 3.9 on your PC and make sure the folder containing python39.dll is in PATH environment variable while starting SQLiteStudio.
Another option is to install any version of Python, install [a Git HEAD build of SQLiteStudio](https://github.com/pawelsalawa/sqlitestudio/actions/runs/10765416239), and configure Python library path in ScriptingPython plugin configuration.
